pysmt/test/test_optimization.py

```diff
@@ -0,0 +1,78 @@
+#
+# This file is part of pySMT.
+#
+#   Copyright 2014 Andrea Micheli and Marco Gario
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+from pysmt.test import TestCase, skipIfNoOptimizerForLogic
+from pysmt.test import main
+
+from pysmt.shortcuts import Optimizer, GE, Int, Symbol, INT, LE, GT, REAL, Real
+from pysmt.shortcuts import And, Plus, Minus, get_env
+from pysmt.logics import QF_LIA, QF_LRA
+
+from pysmt.exceptions import PysmtUnboundedOptimizationError
+
+class TestOptimization(TestCase):
+
+    @skipIfNoOptimizerForLogic(QF_LIA)
+    def test_basic(self):
+        x = Symbol("x", INT)
+        formula = GE(x, Int(10))
+        for oname in get_env().factory.all_optimizers(logic=QF_LIA):
+            with Optimizer(name=oname) as opt:
+                opt.add_assertion(formula)
+                model = opt.optimize(x)
+                self.assertEqual(model[x], Int(10))
+
+    def _auto_satisfy_sua(self, model):
```

Comment:
Isn't this cheating? The function is not checking any condition related to the bounds of the problem, but always raising the exception. A better test would have an unbounded problem and make sure that the Optimizer is able to detect it.

Reply:
The SUA and Incremental algorithms diverge when the objective is unbounded or infinitesimal. I added a predicate to check this "feature" given an optimization algorithm.